src/mscorlib/src/System/String.cs

```diff
@@ -1208,6 +1272,42 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
             }
             return foundCount;
         }
+
+        private struct SplitSeparatorArray
+        {
+            private readonly bool _single;
```

Comment:
Can this field be eliminated if you place constraints on the other two fields? (e.g. _separator != null implies 'single')